maykin_common/otel.py

```diff
@@ -54,6 +55,38 @@
 
 DEFAULT_PROTOCOL: ExportProtocol = "grpc"
 
+# FIXME loading these envvars at module level causes the tests to fail
+# DEFER_SETUP = config(
+#     "_OTEL_DEFER_SETUP",
+#     default=False,
+#     group="Open Telemetry",
+#     help_text=(
+#         "In some situations (similar to uwsgi, see below), initialization must be "
+#         "deferred, e.g. in celery workers with a process pool that fork other "
+#         "processes. Detecting if we're running in a celery master or worker process
+#         "is not obvious, so instead we look at an explicit environment variable."
+#     ),
+# )
+# _ENABLE_RESOURCE_DETECTOR: bool = config(
+#     "_OTEL_ENABLE_CONTAINER_RESOURCE_DETECTOR",
+#     default=False,
+#     group="Open Telemetry",
+#     help_text=(
+#         "Can be enabled to extract additional resource attributes when "
+#         "running in Docker. "
+#         "Don't do this on Kubernetes, as it may lead to conflicting information."
+#     ),
+# )
+# PROTOCOL: ExportProtocol = config(
+#     "OTEL_EXPORTER_OTLP_PROTOCOL",
+#     default=DEFAULT_PROTOCOL,
+#     group="Open Telemetry",
+#     help_text=(
+#         "The name of the protocol to use for OTel, possible options: ``grpc``, "
+#         "``http/protobuf``."
+#     ),
+# )
```

Comment:
I would also not actively document the OTel environment variables, because they are standardized across langauges and SDK by the OTel standard itself. Linking to https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/#general-sdk-configuration should be sufficient. Maintaining a copy of this documentation can only go stale and be wrong eventually, and it does not produce any worthwile benefits IMO.